### ISSUE TITLE
-Wenum-conversion in net/wireless/reg.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#1519]
Initially reported by CI: https://github.com/ClangBuiltLinux/continuous-integration2/runs/4369672398?check_suite_focus=true

```
net/wireless/reg.c:1137:23: error: implicit conversion from enumeration type 'enum nl80211_user_reg_hint_type' to different enumeration type 'enum nl80211_reg_initiator' [-Werror,-Wenum-conversion]
        request->initiator = NL80211_USER_REG_HINT_USER;
                           ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

Inquiry sent: https://lore.kernel.org/r/YaZLEEj2pUU%2FDhZD@archlinux-ax161/

**Comment by nathanchance** [comment]
https://lore.kernel.org/r/YadvTolO8rQcNCd%2F@gimli.kloenk.dev/

**Comment by nathanchance** [comment]
Fixed in `next-20211215`: https://git.kernel.org/jberg/mac80211/c/37d33114240ede043c42463a6347f68ed72d6904
